fs-storage/src/file_storage.rs

```diff
@@ -224,13 +234,54 @@ where
     }
 }
 
+/// Combine two `FileStorage` instances into a new instance.
+pub fn combine_file_storages<K, V>(
+    a: &FileStorage<K, V>,
+    b: &FileStorage<K, V>,
+) -> FileStorage<K, V>
+where
+    K: Ord + Clone,
+    V: Clone + PartialOrd + Default,
+{
+    let mut combined_entries = BTreeMap::new();
+
+    for (key, value) in &a.data.entries {
+        combined_entries.insert(key.clone(), value.clone());
+    }
+
+    for (key, value) in &b.data.entries {
+        if let Some(existing_value) = combined_entries.get(key) {
+            let resolved_value = if value > existing_value {
+                value.clone()
+            } else {
+                existing_value.clone()
+            };
+            combined_entries.insert(key.clone(), resolved_value);
+        } else {
+            combined_entries.insert(key.clone(), value.clone());
+        }
+    }
+
+    FileStorage {
+        label: a.label.clone(),
+        path: a.path.clone(),
+        modified: SystemTime::now(),
+        data: FileStorageData {
+            version: STORAGE_VERSION,
+            entries: combined_entries,
```

Comment:
The `combined_entries` also need to be written to disk, otherwise it will be out of sync with data on disk.